test/fork/migration-v2.js

```diff
@@ -689,143 +681,178 @@ describe('V2 upgrade', function () {
+
+    // Nexus Mutual Foundation Pool
+    console.log('Nexus Mutual Foundation Pool');
+    const foundationPoolId = 0;
+    // The entire NXM balance is unlocked and sent back to the Foundation
+    // TODO: Needs some Solidity changes to be able to fully migrate them as well
+    expect(depositsInStakingPools[foundationPoolId]).to.be.equal(0); // depositInPS[FOUNDATION]
+    expect(nxmBalancesAfter[FOUNDATION]).to.be.equal(nxmBalancesBefore[FOUNDATION].add(depositInPS[FOUNDATION]));
+
+    // Hugh Pool
+    console.log('Hugh Pool');
+    const hughPoolId = 1;
+    precisionTolerance = 1; // 1 wei
+    expect(depositsInStakingPools[hughPoolId]).to.be.equal(depositInPS[HUGH].sub(precisionTolerance));
+    // No NXM gets unlocked, so the balance shouldn't change
+    precisionTolerance = 1; // 1 wei
+    expect(nxmBalancesBefore[HUGH]).to.be.equal(nxmBalancesAfter[HUGH].sub(precisionTolerance));
+
+    // Armor AAA Pool
+    console.log('Armor AAA Pool');
+    const armorAAAPoolId = 2;
+    // 5% of the AAA allocation must be unlocked
+    const expectedArmorAAAPoolBalance = depositInPS[ARMOR].mul(75).div(100).mul(95).div(100);
+    precisionTolerance = 1;
+    expect(depositsInStakingPools[armorAAAPoolId]).to.be.equal(expectedArmorAAAPoolBalance.sub(precisionTolerance));
+
+    // Armor AA Pool
+    console.log('Armor AA Pool');
+    const armorAAPoolId = 3;
+    // 5% of the AA allocation must be unlocked
+    const expectedArmorAAPoolBalance = depositInPS[ARMOR].mul(25).div(100).mul(95).div(100);
+    precisionTolerance = 1;
+    expect(depositsInStakingPools[armorAAPoolId]).to.be.equal(expectedArmorAAPoolBalance.sub(precisionTolerance));
+
+    // Overall we must unlock 5% of Armor's total tokens in PS
+    precisionTolerance = 4;
+    expect(nxmBalancesAfter[ARMOR]).to.be.equal(
+      nxmBalancesBefore[ARMOR].add(depositInPS[ARMOR].mul(5).div(100)).add(precisionTolerance),
     );
 
-    // assert deposits are now 0 for each staker
+    // Check PS deposits are now 0 for all selected stakers
+    console.log('Checking PS deposits are now 0 for all selected stakers');
     await Promise.all(
       topStakers.map(async staker => {
         const deposit = await this.pooledStaking.stakerDeposit(staker);
         expect(deposit).to.be.equal(0);
       }),
     );
 
-    // Assert deposit for Armor Pool 0
-    const v2ProductAddresses = require(path.join(
-      config.paths.root,
-      'scripts/v2-migration/products/output/product-addresses.json',
-    ));
+    // Check staked products prices for all staking pools
+    console.log('Checking staked products prices for all staking pools');
+
+    const productAddresses = PRODUCT_ADDRESSES_OUTPUT.map(address => address.toLowerCase());
+    const stakers = topStakers.concat([ARMOR]); // Armor has 2 pools - do this so we can iterate below
+
+    const deprecatedProducts = new Set();
+    const productsWithNoStake = new Set();
+
+    for (let poolId = 0; poolId < stakingPoolCount; poolId++) {
```

Comment:
feels like it's easier if inner loop here stays a function and is pulled outside 


the block is already very large and it would make it easier to read

Reply:
I'd prefer to be able to see the code in one place, rather than scroll up and down in this case